health-services/individual/src/main/java/org/egov/individual/service/IndividualService.java

```diff
@@ -406,39 +400,47 @@ public void putInCache(List<Individual> individuals) {
         log.info("successfully put individuals in cache");
     }
 
-    private void integrateWithUserService(IndividualBulkRequest request,
-                                          List<Individual> encryptedIndividualList, ApiOperation apiOperation) {
+    private List<Individual> integrateWithUserService(IndividualBulkRequest request,
+                                          List<Individual> individualList, ApiOperation apiOperation,
+                                          Map<Individual, ErrorDetails> errorDetails) {
+        List<Individual> validIndividuals = new ArrayList<>(individualList);
         if (properties.isUserSyncEnabled()) {
-            try {
-                if (apiOperation.equals(ApiOperation.UPDATE)) {
-                    userIntegrationService.updateUser(encryptedIndividualList,
-                            request.getRequestInfo());
-                    log.info("successfully updated user for {} individuals",
-                            encryptedIndividualList.size());
-                } else if (apiOperation.equals(ApiOperation.CREATE)) {
-                    List<UserRequest> userRequests = userIntegrationService.createUser(encryptedIndividualList,
-                            request.getRequestInfo());
-                    for (int i = 0; i < encryptedIndividualList.size(); i++) {
-                        if (Boolean.TRUE.equals(encryptedIndividualList.get(i).getIsSystemUser())) {
-                            encryptedIndividualList.get(i)
-                                    .setUserId(Long.toString(userRequests.get(i).getId()));
-                            encryptedIndividualList.get(i).setUserUuid(userRequests.get(i).getUuid());
+            for (Individual individual : individualList) {
```

Comment:
User service doesn't support bulk creation, so anyways it was getting streamed and called inside this 'createUser' function. If we don't use for loop then even if anyone of the user is not created then other users for valid individuals will also not be created. This for loop is inevitable to create errorDetails in catch block and maintain valid individuals. I have kept it here and not transferred to 'createUser' because other wise we have to return all three objects( errorDetails, validIndividuals and createdUsers ) back here.